### PR TITLE
Add RHEL 7 test flow to automation

### PR DESCRIPTION
**Pull request by eranha**
### What does this PR do?
As part of automating our tests, we need to add the ability to run our tests on RHEL 7. 
Added new files:

- RHEL 7 dockerfile (downloads and compiles Python3.9 make all libs available for pyinstaller)
- Centos repo files
- repo public keyfiles (for GPG check)
- Turned on RHEL 8 repo GPG check

### What ticket does this PR close?
[https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-11327](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-11327)

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: test/RPM-GPG-KEY-CentOS-7 [hunk 1]
```diff
@@ -0,0 +1,30 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
```
Comment (sigalsax) [inline test/RPM-GPG-KEY-CentOS-7:1]: we shouldn't be pushing keys even if they are public @eranha

File: test/RPM-GPG-KEY-CentOS-Official [hunk 2]
```diff
@@ -0,0 +1,30 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
```
Comment (sigalsax) [inline test/RPM-GPG-KEY-CentOS-Official:1]: same comment applies here https://github.com/cyberark/conjur-api-python3/pull/336/files#r707068805

File: Jenkinsfile [hunk 3]
```diff
@@ -31,6 +31,7 @@ pipeline {
       }
       post {
         always {
+              echo 'dummy'
```
Comment (sigalsax) [inline Jenkinsfile:34]: delete

File: Jenkinsfile [hunk 4]
```diff
@@ -50,9 +51,21 @@ pipeline {
...
+        always {
+          junit 'output/**/*.xml'
+        }
+      }
+    }
+
     stage('RHEL8 Integration tests') {
       steps {
-        sh 'summon -e common ./bin/test_integration_rhel8'
+        sh 'summon -e common ./bin/test_integration_rhel test-rhel8'
```
Comment (sigalsax) [inline Jenkinsfile:68]: can just be
```suggestion
        sh 'summon -e common ./bin/test_integration_rhel rhel8'
```
IMO

File: Dockerfile.test.rhel7 [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                                            jq \
+                                            libffi-devel \
+                                            gnome-keyring \
+                                            procps \
+                                            zlib-devel \
+                                            wget \
+                                            openssl-devel \
+                                            bzip2-devel \
+                                            dbus-x11 \
+                                            && yum --disableplugin=subscription-manager clean all
```
Comment (orenbm) [inline Dockerfile.test.rhel7:31]: I would change the indentation to this:
```suggestion
RUN yum \
    --disableplugin=subscription-manager \
    install -y \
      bash \
      wget \
      binutils \
      yum-utils \
      gcc gcc-c++ make \
      curl \
      jq \
      libffi-devel \
      gnome-keyring \
      procps \
      zlib-devel \
      wget \
      openssl-devel \
      bzip2-devel \
      dbus-x11 \
    && yum --disableplugin=subscription-manager clean all \
```

Comment (orenbm) [reply Dockerfile.test.rhel7:31]: if you think it's better please do this for the rest of the file

File: Dockerfile.test.rhel7 [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+RUN ldconfig /usr/local/lib
+ENV PATH=${PATH}:/usr/local/lib
+RUN pip3.9 install wheel
+RUN pip3.9 install -r requirements.txt
+COPY ./bin/build_integrations_tests_runner /build_integrations_tests_runner
+COPY . $INSTALL_DIR
+COPY ./test/configure_test_executor.sh /configure_test_executor.sh
+
+ENTRYPOINT ["./test/configure_test_executor.sh"]
+
```
Comment (orenbm) [inline Dockerfile.test.rhel7:63]: remove this empty line please

File: bin/test_integration_rhel [hunk 7]
```diff
@@ -1,23 +1,32 @@
 #!/bin/bash -ex
 
+CONTAINER_ID="$1"
```
Comment (orenbm) [inline bin/test_integration_rhel:3]: now that we have more than one input, can we change this to pure flags? 

something like this will do:
```
while true ; do
  case "$1" in
    --rhel-version=* ) RHEL_VERSION="${option#*=}" ; shift ;;
    --debug ) DEBUG="true" ; shift ;;
    --help ) print_help ; shift ;;
     * ) if [ -z "$1" ]; then break; else echo "$1 is not a valid option"; exit 1; fi;;
  esac
done
```

Note that i changed `CONTAINER_ID` to `RHEL_VERSION` to be more accurate (as i understand it) and that i added a `print_help` call that you may implement if you like. something like this will do (but change the data):
```
function print_help() {
  cat << EOF

Creates an Appliance release

Usage: start [options]

    -h, --help        Shows this help message.

    -r, --revert      Removes the current version tags. This allows the appliance
                      to be re-released.

    -f, --force       Release even if there are no changes. This is necessary
                      after a revert.

EOF
}
```

Comment (orenbm) [reply bin/test_integration_rhel:3]: regarding the `CONTAINER_ID` being `test-rhel7/8`, i think it is an implementation detail. i would get `RHEL_VERSION=7/8` as input and set `CONTAINER_ID="test-rhel${RHEL_VERSION}` once we get it from the user.

File: bin/test_integration_rhel [hunk 8]
```diff
@@ -1,23 +1,32 @@
...
   DEBUG="true"
   shift
 fi
 
+if [[ -z "${CONTAINER_ID}" ]]; then
+  echo "RHEL container not set using 'test-rhel8' as default"
+  CONTAINER_ID="test-rhel8"
+fi
+
+echo "Test integration on ${CONTAINER_ID}"
```
Comment (orenbm) [inline bin/test_integration_rhel:22]:
```suggestion
echo "Run integration tests on RHEL version ${RHEL_VERSION}"
```

File: docker-compose.yml [hunk 9]
```diff
@@ -111,3 +111,24 @@ services:
...
+      TEST_ENV: "True"
+      SERVER_MODE: Null
+      LOGIN: Null
+      ADMIN_PASSWORD: Null
+      TEST_HOSTNAME: Null
+      ACCOUNT: Null
+    depends_on:
+      - conjur-appliance
+    volumes:
+      - ${PWD}:/opt/conjur-api-python3
```
Comment (orenbm) [inline docker-compose.yml:134]: missing new line a EOF

File: test/CentOS-7-Linux-AppStream.repo [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
+
+#additional packages that extend functionality of existing packages
+[centosplus]
+name=CentOS-7 - Plus
+#mirrorlist=http://mirrorlist.centos.org/?release=7&arch=$basearch&repo=centosplus&infra=$infra
+baseurl=http://mirror.centos.org/centos/7/centosplus/$basearch/
+gpgcheck=1
+enabled=0
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
```
Comment (orenbm) [inline test/CentOS-7-Linux-AppStream.repo:31]: missing new line a EOF

File: test/CentOS-Linux-AppStream.repo [hunk 11]
```diff
@@ -7,5 +7,6 @@
 name=CentOS Linux $releasever - AppStream
 mirrorlist=http://mirrorlist.centos.org/?release=$releasever&arch=$basearch&repo=AppStream&infra=$infra
 #baseurl=http://mirror.centos.org/$contentdir/$releasever/AppStream/$basearch/os/
-gpgcheck=0
-enabled=1
\ No newline at end of file
+gpgcheck=1
+enabled=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS
```
Comment (orenbm) [inline test/CentOS-Linux-AppStream.repo:12]: missing new line a EOF

**Review by orenbm — CHANGES_REQUESTED** [review]
left a few comments

File: Dockerfile.test.rhel7 [hunk 12]
```diff
@@ -0,0 +1,63 @@
+FROM registry.access.redhat.com/ubi7/ubi
```
Comment (sigalsax) [inline Dockerfile.test.rhel7:1]: Can you please add a comment at the top of the file explaining what this file does and is trying to accomplish (configure dbus for example). It has a look of good and complex stuff in here

File: Dockerfile.test.rhel7 [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+                                            zlib-devel \
+                                            wget \
+                                            openssl-devel \
+                                            bzip2-devel \
+                                            dbus-x11 \
+                                            && yum --disableplugin=subscription-manager clean all
+
+
+# install python 3.9
+ENV PYTHON_CONFIGURE_OPTS="--enable-shared"
```
Comment (sigalsax) [inline Dockerfile.test.rhel7:35]: Comment why we need this?

File: Dockerfile.test.rhel7 [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+
+# configure dbus-11
+RUN dbus-uuidgen > /var/lib/dbus/machine-id
+
+RUN mkdir -p $INSTALL_DIR
+WORKDIR $INSTALL_DIR
+COPY ./requirements.txt $INSTALL_DIR/
+RUN yum --disableplugin=subscription-manager install -y rh-python38-python-devel \
+                                                        centos-release-scl \
+                                                        devtoolset-7 \
```
Comment (sigalsax) [inline Dockerfile.test.rhel7:49]: Extra lines here

File: Dockerfile.test.rhel7 [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+RUN scl enable devtoolset-7 bash
+RUN cp /usr/local/bin/python3.9 /usr/local/bin/python3
+ENV PATH=/usr/local/bin:${PATH}
+RUN ldconfig /usr/local/lib
+ENV PATH=${PATH}:/usr/local/lib
+RUN pip3.9 install wheel
+RUN pip3.9 install -r requirements.txt
+COPY ./bin/build_integrations_tests_runner /build_integrations_tests_runner
+COPY . $INSTALL_DIR
+COPY ./test/configure_test_executor.sh /configure_test_executor.sh
```
Comment (sigalsax) [inline Dockerfile.test.rhel7:60]: Pleas work to separate the COPYs RUNs etc by a new line. Otherwise it’s hard to digest what is happening here

File: bin/test_integration_rhel (hunk 7 above)
Comment (sigalsax) [reply bin/test_integration_rhel:3]: Plus plus on this

File: test/CentOS-7-Linux-AppStream.repo [hunk 16]
```diff
@@ -0,0 +1,31 @@
+[base]
+name=CentOS-7 - Base
+#mirrorlist=http://mirrorlist.centos.org/?release=7&arch=$basearch&repo=os&infra=$infra
```
Comment (sigalsax) [inline test/CentOS-7-Linux-AppStream.repo:3]: You can remove this since it’s in a comment no?

File: test/CentOS-Linux-AppStream.repo (hunk 11 above)
Comment (sigalsax) [reply test/CentOS-Linux-AppStream.repo:12]: Why is gpgcheck updated?

File: Dockerfile.test.rhel7 (hunk 14 above)
Comment (orenbm) [reply Dockerfile.test.rhel7:66]: please align as above

File: Dockerfile.test.rhel7 [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+RUN ldconfig /usr/local/lib
+ENV PATH=${PATH}:/usr/local/lib
+RUN pip3.9 install wheel
+RUN pip3.9 install -r requirements.txt
+
+COPY ./bin/build_integrations_tests_runner /build_integrations_tests_runner
+COPY . $INSTALL_DIR
+COPY ./test/configure_test_executor.sh /configure_test_executor.sh
+
+ENTRYPOINT ["./test/configure_test_executor.sh"]
```
Comment (orenbm) [inline Dockerfile.test.rhel7:80]: missing new line at EOF

File: Dockerfile.test.rhel7 [hunk 18]
```diff
@@ -59,26 +59,31 @@ RUN wget https://www.python.org/ftp/python/3.9.6/Python-3.9.6.tgz \
     && ./Python-3.9.6/configure --enable-shared \
     && make altinstall
 
-# configure dbus-11
+# Configure generate unique machne-id file required by dbus-11
```
Comment (orenbm) [inline Dockerfile.test.rhel7:62]: ?
```suggestion
# Generate unique machne-id file required by dbus-11
```

File: test/CentOS-7-Linux-AppStream.repo (hunk 16 above)
Comment (eranha) [reply test/CentOS-7-Linux-AppStream.repo:3]: That's a file copied from centos

File: test/CentOS-Linux-AppStream.repo (hunk 11 above)
Comment (eranha) [reply test/CentOS-Linux-AppStream.repo:12]: Because until now there was no gpg check in 8

File: test/RPM-GPG-KEY-CentOS-7 (hunk 1 above)
Comment (eranha) [reply test/RPM-GPG-KEY-CentOS-7:1]: Talked to @sashaCher is ok with that, these are public keys, the alternative is to download it online, if a URL changes it would break the build

File: .idea/.gitignore [hunk 19]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
+/shelf/
+/workspace.xml
+# Datasource local storage ignored files
+/dataSources/
+/dataSources.local.xml
+# Editor-based HTTP Client requests
+/httpRequests/
```
Comment (sigalsax) [inline .idea/.gitignore:8]: do we push any of these files currently? If so we might need to remove them from the remote

File: Dockerfile.test.rhel7 [hunk 20]
```diff
@@ -0,0 +1,85 @@
+# Dockerfile.test.rhel
+#
+# Dockerfile for running tests on RHEL7. To install Python3.9 no RHEL
```
Comment (sigalsax) [inline Dockerfile.test.rhel7:3]: "Python3.9 no RHEL"? What does that mean?

File: Dockerfile.test.rhel7 [hunk 21]
```diff
@@ -0,0 +1,85 @@
...
+# The Centos repo files enables access to RPMs required by pip and PyInstaller
+# that are not available on the RHEL7 built-in repos.
+
+# Base image is RHEL7
+FROM registry.access.redhat.com/ubi7/ubi
+
+# Set the install dir env param
+ENV INSTALL_DIR=/opt/conjur-api-python3
+
+# Copy below repo files to tum repo dor to make rpms available
```
Comment (sigalsax) [inline Dockerfile.test.rhel7:14]: tum? Yum?

```suggestion
# Copy below repo files to yum repo directory to make rpms available
```

File: Dockerfile.test.rhel7 [hunk 22]
```diff
@@ -0,0 +1,85 @@
...
+# Configure generate unique machne-id file required by dbus-11
+RUN dbus-uuidgen > /var/lib/dbus/machine-id
+
+RUN mkdir -p $INSTALL_DIR
+WORKDIR $INSTALL_DIR
+
+# Copy Python requirements file
+COPY ./requirements.txt $INSTALL_DIR/
+
+# Install Pthon 3.9.7 using pyenv, wheel and required libs
```
Comment (sigalsax) [inline Dockerfile.test.rhel7:66]:
```suggestion
# Install Python 3.9.7 using pyenv, wheel and required libs
```

File: Dockerfile.test.rhel7 [hunk 23]
```diff
@@ -0,0 +1,85 @@
...
+    && pyenv install 3.9.7 \
+    && pyenv global 3.9.7 \
+    && pip install wheel \
+    && pip install -r requirements.txt
+
+# Copy script files
+COPY ./bin/build_integrations_tests_runner ./test/configure_test_executor.sh /
+COPY . $INSTALL_DIR
+
+# Required to build PyInstaller
```
Comment (sigalsax) [inline Dockerfile.test.rhel7:81]: Are we building pyinstaller or using?

File: Dockerfile.test.rhel8 [hunk 24]
```diff
@@ -26,15 +32,15 @@ RUN yum --disableplugin=subscription-manager -y \
 RUN mkdir -p $INSTALL_DIR
 WORKDIR $INSTALL_DIR
 
+# Configure generate unique machne-id file required by dbus-11
```
Comment (sigalsax) [inline Dockerfile.test.rhel8:35]:
```suggestion
# Generate unique machine-id file required by dbus-11
```

File: Jenkinsfile [hunk 25]
```diff
@@ -16,48 +16,64 @@ pipeline {
     stage('Linting') {
       parallel {
         stage('Code') {
-          steps { sh './bin/test_linting' }
+          steps { echo './bin/test_linting' }
         }
 
         stage('Changelog') {
-          steps { sh './bin/test_changelog' }
+          steps { echo './bin/test_changelog' }
```
Comment (sigalsax) [inline Jenkinsfile:23]: curious, why this change?

File: Jenkinsfile [hunk 26]
```diff
@@ -16,48 +16,64 @@ pipeline {
...
       steps {
-        sh './bin/test_unit'
+        echo './bin/test_unit'
       }
       post {
         always {
-          junit 'output/**/*.xml'
-          cobertura autoUpdateHealth: false, autoUpdateStability: false, coberturaReportFile: 'coverage.xml', conditionalCoverageTargets: '50, 0, 50', failUnhealthy: true, failUnstable: false, lineCoverageTargets: '50, 0, 50', maxNumberOfBuilds: 0, methodCoverageTargets: '50, 0, 50', onlyStable: false, sourceEncoding: 'ASCII', zoomCoverageChart: false
-          ccCoverage("coverage.py")
+      echo 'dummy'
```
Comment (sigalsax) [inline Jenkinsfile:34]: comment back in? and remove 'dummy'?

File: Jenkinsfile [hunk 27]
```diff
@@ -16,48 +16,64 @@ pipeline {
...
       steps {
-        sh './bin/test_integration'
+        echo './bin/test_integration'
       }
 
       post {
         always {
-          junit 'output/**/*.xml'
+              echo 'dummy'
+          //junit 'output/**/*.xml'
```
Comment (sigalsax) [inline Jenkinsfile:50]: here too comment back in and remove dummy?

File: Jenkinsfile [hunk 28]
```diff
@@ -16,48 +16,64 @@ pipeline {
...
+        }
+      }
+    }
+
+    stage('RHEL7 Integration tests') {
+      steps {
+        sh 'summon -e common ./bin/test_integration_rhel --rhel-version=7'
```
Comment (sigalsax) [inline Jenkinsfile:57]: if we are following the pattern of using 'echo'
```suggestion
        echo 'summon -e common ./bin/test_integration_rhel --rhel-version=7'
```

File: Jenkinsfile [hunk 29]
```diff
@@ -16,48 +16,64 @@ pipeline {
...
+      }
+
+      post {
+        always {
+        echo 'dummy'
```
Comment (sigalsax) [inline Jenkinsfile:62]: remove and comment the below back in?

File: Jenkinsfile [hunk 30]
```diff
@@ -16,48 +16,64 @@ pipeline {
...
       steps {
-        sh 'summon -e common ./bin/test_integration_rhel8'
+        echo 'summon -e common ./bin/test_integration_rhel --rhel-version=8'
       }
 
       post {
         always {
-          junit 'output/**/*.xml'
+              echo 'dummy'
+          //junit 'output/**/*.xml'
```
Comment (sigalsax) [inline Jenkinsfile:76]: uncomment and remove 'dummy'?

File: bin/build_integrations_tests_runner [hunk 31]
```diff
@@ -2,14 +2,16 @@
 
 function _build_test_integration_exec() {
   echo "Dropping into virtual environment..."
-  python3 -m venv venv
+  python -m venv venv
```
Comment (sigalsax) [inline bin/build_integrations_tests_runner:5]: i still think we should have `python3` even though we define python 3.9

File: bin/build_integrations_tests_runner [hunk 32]
```diff
@@ -2,14 +2,16 @@
...
   source venv/bin/activate
 
   echo "Installing requirements for tests..."
-  pip3 install -r requirements.txt
+  echo "@@@ build_test_integration_exec PATH=${PATH}"
```
Comment (sigalsax) [inline bin/build_integrations_tests_runner:9]: what is this for?

File: bin/test_integration_rhel [hunk 33]
```diff
@@ -1,23 +1,55 @@
...
+    --rhel-version=*  Run in RHEL container specific version (7|8) default OS version is 8.
+
+    -d, --debug           Step into the container.
+
+EOF
+}
+
+while true ; do
+  case "$1" in
+    #--rhel-version=* ) RHEL_VERSION="${option#*=}" ; shift ;;
```
Comment (sigalsax) [inline bin/test_integration_rhel:21]: remove?

File: bin/test_integration_rhel [hunk 34]
```diff
@@ -1,23 +1,55 @@
...
 trap 'echo "ERROR: Test script encountered an error!"; docker-compose logs &> cleanup.log; cleanup' ERR
 trap 'cleanup' EXIT
 cleanup
 
+if [[ "${RHEL_VERSION}" = "8" ]]; then
+  CONTAINER_ID="rhel8"
+elif [[ "${RHEL_VERSION}" = "7" ]]; then
+  CONTAINER_ID="rhel7"
+else
+  echo "RHEL container not set using 'rhel8' as default"
```
Comment (sigalsax) [inline bin/test_integration_rhel:45]:
```suggestion
  echo "RHEL container not set. Using 'rhel8' as default..."
```

File: cleanup.log [hunk 35]
```diff
@@ -0,0 +1 @@
+Attaching to
```
Comment (sigalsax) [inline cleanup.log:1]: do we need this?

File: .idea/.gitignore (hunk 19 above)
Comment (eranha) [reply .idea/.gitignore]: By mistake, removed.

File: Dockerfile.test.rhel7 (hunk 20 above)
Comment (eranha) [reply Dockerfile.test.rhel7:3]: on

File: Dockerfile.test.rhel7 (hunk 23 above)
Comment (eranha) [reply Dockerfile.test.rhel7:81]: From what I saw in the log we're building it.

File: Jenkinsfile (hunk 25 above)
Comment (eranha) [reply Jenkinsfile:23]: temporary to speed up the build

File: bin/build_integrations_tests_runner (hunk 31 above)
Comment (eranha) [reply bin/build_integrations_tests_runner:5]: I'm using pyenv and you set the Pyhton to the desired version, you can simply call python

File: Dockerfile.test.rhel8 [hunk 36]
```diff
@@ -26,15 +32,16 @@ RUN yum --disableplugin=subscription-manager -y \
 RUN mkdir -p $INSTALL_DIR
 WORKDIR $INSTALL_DIR
 
+# Generate unique machne-id file required by dbus-11
+RUN dbus-uuidgen > /var/lib/dbus/machine-id
+
+
```
Comment (orenbm) [inline Dockerfile.test.rhel8:38]: you can use only one empty line

File: bin/build_integrations_tests_runner [hunk 37]
```diff
@@ -2,14 +2,15 @@
...
 
   echo "Installing requirements for tests..."
-  pip3 install -r requirements.txt
+  pip install -r requirements.txt
 
   pyinstaller -F /opt/conjur-api-python3/test/util/test_runners/integrations_tests_runner.py
   deactivate
 }
 
+
```
Comment (orenbm) [inline bin/build_integrations_tests_runner:15]: redundant empty line

File: bin/test_integration_rhel [hunk 38]
```diff
@@ -1,23 +1,57 @@
...
+
+Run integration tests on RHEL container
+
+Usage: test_integration_rhel [options]
+
+    -h, --help            Shows this help message.
+
+    --rhel-version=7  Run in RHEL 7 container.
+
+    --rhel-version=8  Run in RHEL 8 container.
```
Comment (orenbm) [inline bin/test_integration_rhel:14]: fix indentation. also - i think we can have only an entry for `--rhel-version` and say that it is used to `Set the RHEL version to test (e.g --rhel-version=7)`

File: bin/test_integration_rhel [hunk 39]
```diff
@@ -1,23 +1,57 @@
...
+
+    -d, --debug           Step into the container.
+
+EOF
+}
+
+while true ; do
+  case "$1" in
+    #--rhel-version=* ) RHEL_VERSION="${option#*=}" ; shift ;;
```
Comment (orenbm) [inline bin/test_integration_rhel:23]: can you explain please how come [this](https://github.com/conjurinc/dap-package/blob/3a980a45792aac85cd278ad3f520617b020154cd/lib/user_interactions.sh#L136) works but in your script it doesn't?

File: test/CentOS-7-Linux-AppStream.repo (hunk 16 above)
Comment (orenbm) [reply test/CentOS-7-Linux-AppStream.repo:3]: i think that you can't remove these lines and they are not treated as comments

File: Dockerfile.test.rhel7 (hunk 23 above)
Comment (sigalsax) [reply Dockerfile.test.rhel7:81]: which log and where?

File: Dockerfile.test.rhel8 [hunk 40]
```diff
@@ -26,15 +32,16 @@ RUN yum --disableplugin=subscription-manager -y \
...
 COPY ./requirements.txt $INSTALL_DIR/
 RUN pip3 install wheel && \
     pip3 install -r requirements.txt
 
-COPY ./bin/build_integrations_tests_runner /build_integrations_tests_runner
-
+COPY ./bin/build_integrations_tests_runner ./test/configure_test_executor.sh /
```
Comment (sigalsax) [inline Dockerfile.test.rhel8:43]: what is the final `/` at the end? Isn't it COPY <source> <target> ?

File: Jenkinsfile (hunk 25 above)
Comment (sigalsax) [reply Jenkinsfile:23]: Are you done with the implementation? If so and only minor tweaks remain, I would switch these back

File: bin/test_integration_rhel [hunk 41]
```diff
@@ -1,23 +1,57 @@
...
+  esac
+done
+
 cleanup() {
   echo "Cleaning up..."
   echo "Cleaning up... logs in cleanup.log"
   docker-compose rm --stop --force
 }
 
-DEBUG="false"
```
Comment (sigalsax) [inline bin/test_integration_rhel:9]: why did we remove this? this is important

File: test/CentOS-Linux-AppStream.repo [hunk 42]
```diff
@@ -7,5 +7,6 @@
 name=CentOS Linux $releasever - AppStream
 mirrorlist=http://mirrorlist.centos.org/?release=$releasever&arch=$basearch&repo=AppStream&infra=$infra
 #baseurl=http://mirror.centos.org/$contentdir/$releasever/AppStream/$basearch/os/
-gpgcheck=0
-enabled=1
\ No newline at end of file
+gpgcheck=1
```
Comment (sigalsax) [inline test/CentOS-Linux-AppStream.repo:10]: why do we need a gpg check btw? why do we need gpg at all?

**Review by sigalsax — CHANGES_REQUESTED** [review]
Left some final comments but otherwise looks like a job well done!

I am blocking the PR from being merged so just to be sure we don't forget to switch back the 'echo' -> 'sh' in the Jenkinsfile

BTW, I checked the test run in the pipeline and all looks good. The tests are running as they should :) @eranha

File: Dockerfile.test.rhel7 [hunk 43]
```diff
@@ -0,0 +1,85 @@
...
+      devtoolset-7 \
+      && yum --disableplugin=subscription-manager clean all
+
+# Generate unique machne-id file required by dbus-11
+RUN dbus-uuidgen > /var/lib/dbus/machine-id
+
+RUN mkdir -p $INSTALL_DIR
+WORKDIR $INSTALL_DIR
+
+# Copy Python requirements file
```
Comment (sigalsax) [inline Dockerfile.test.rhel7:63]: Small nit but no biggy

```suggestion
# Copy project's requirements file
```

File: Dockerfile.test.rhel7 (hunk 23 above)
Comment (sigalsax) [reply Dockerfile.test.rhel7:81]: @eranha

File: Dockerfile.test.rhel8 (hunk 36 above)
Comment (sigalsax) [reply Dockerfile.test.rhel8:38]: @eranha

File: Jenkinsfile (hunk 25 above)
Comment (sigalsax) [reply Jenkinsfile:23]: @eranha

File: bin/build_integrations_tests_runner (hunk 37 above)
Comment (sigalsax) [reply bin/build_integrations_tests_runner:15]: @eranha

File: bin/test_integration_rhel (hunk 38 above)
Comment (sigalsax) [reply bin/test_integration_rhel:14]: @eranha  ^

File: bin/test_integration_rhel (hunk 33 above)
Comment (sigalsax) [reply bin/test_integration_rhel:23]: @eranha

File: bin/test_integration_rhel (hunk 41 above)
Comment (sigalsax) [reply bin/test_integration_rhel:9]: @eranha

File: test/CentOS-Linux-AppStream.repo (hunk 42 above)
Comment (sigalsax) [reply test/CentOS-Linux-AppStream.repo:10]: @eranha

File: bin/test_integration_rhel (hunk 39 above)
Comment (eranha) [reply bin/test_integration_rhel:23]: Because I did:
 `case "$1" in` and in the link you've sent it's  `case "$@" in` and it does work.

Comment (eranha) [reply bin/test_integration_rhel:23]: Changed

File: Dockerfile.test.rhel8 (hunk 40 above)
Comment (eranha) [reply Dockerfile.test.rhel8:43]: @sigalsax I copy multiple files to the root

File: test/CentOS-Linux-AppStream.repo (hunk 42 above)
Comment (eranha) [reply test/CentOS-Linux-AppStream.repo:10]: @sigalsax To verify the downloaded RPMs

File: bin/test_integration_rhel [hunk 44]
```diff
@@ -1,23 +1,59 @@
...
+function print_help() {
+  cat << EOF
+
+Run integration tests on RHEL container
+
+Usage: test_integration_rhel [options]
+
+    -h, --help            Shows this help message.
+
+    --rhel-version=7      Used to Set the RHEL version to test (e.g --rhel-version=7).
```
Comment (orenbm) [inline bin/test_integration_rhel:12]: i think that the flag name should be alone and the example should go at the end (like you already have)
```suggestion
    --rhel-version      Used to Set the RHEL version to test (e.g --rhel-version=7).
```

File: bin/test_integration_rhel (hunk 41 above)
Comment (orenbm) [reply bin/test_integration_rhel:9]: @sigalsax it is now written in line 24. or do we still need to explicitly set this to FALSE?

File: bin/test_integration_rhel (hunk 44 above)
Comment (orenbm) [reply bin/test_integration_rhel:12]: also - it seems that the default value is 8. can you add this to this comment?

File: bin/test_integration_rhel [hunk 45]
```diff
@@ -53,7 +89,9 @@ function _configure_conjur() {
   _cmd="$_cmd --hostname $TEST_HOSTNAME"
   _cmd="$_cmd --master-altnames $TEST_HOSTNAME"
   _cmd="$_cmd --admin-password $ADMIN_PASSWORD"
-  _cmd="$_cmd dev"
+  _cmd="$_cmd ${ACCOUNT}"
```
Comment (orenbm) [inline bin/test_integration_rhel:92]: should `ACCOUNT` be set to `test` instead?

**Review by orenbm — CHANGES_REQUESTED** [review]
left some comments + you have some old ones waiting

File: bin/test_integration_rhel [hunk 46]
```diff
@@ -1,23 +1,59 @@
...
+function print_help() {
+  cat << EOF
+
+Run integration tests on RHEL container
+
+Usage: test_integration_rhel [options]
+
+    -h, --help            Shows this help message.
+
+    --rhel-version=7      Used to Set the RHEL version to test (e.g --rhel-version=7). Default version is 8.
```
Comment (orenbm) [inline bin/test_integration_rhel:12]: i would change `--rhel-version=7` to `--rhel-version` or `--rhel-version=<version>`. whichever you think is better. The example is already given

Comment (orenbm) [reply bin/test_integration_rhel:12]: also change `Set` to `set`

File: bin/test_integration_rhel (hunk 45 above)
Comment (eranha) [reply bin/test_integration_rhel:92]: @orenbm I think it should be set form the docker-compose we just ignored it

Comment (orenbm) [reply bin/test_integration_rhel:92]: i don't see it set in `docker-compose` but in line 68 of this file. i would change it there.

File: conjur/argument_parser/parser_utils.py [hunk 47]
```diff
@@ -19,6 +19,7 @@ def header(*args) -> str:
     """
     This method builds the header for the main screen.
     """
+    # pylint: disable = consider-using-f-string
```
Comment (orenbm) [inline conjur/argument_parser/parser_utils.py:22]: please rebase on master. @sigalsax already fixed it

File: bin/test_integration_rhel (hunk 45 above)
Comment (eranha) [reply bin/test_integration_rhel:92]: it is set in test_integration_rhel you can override it in docker-compose the change I did was not to ignore the ACCOUNT var

File: conjur/wrapper/http_wrapper.py [hunk 48]
```diff
@@ -55,7 +55,7 @@ def invoke_endpoint(http_verb: HttpVerb, endpoint: ConjurEndpoint, params: dict,
     headers = {}
     if api_token:
         encoded_token = base64.b64encode(api_token.encode()).decode('utf-8')
-        headers['Authorization'] = 'Token token="{}"'.format(encoded_token)
+        headers['Authorization'] = f'Token token="{encoded_token}"'
```
Comment (orenbm) [inline conjur/wrapper/http_wrapper.py:58]: is this related to this PR? if not - can you please rearrange the commits to show that?

Comment (eranha) [reply conjur/wrapper/http_wrapper.py:58]: Did it with Matan to fix the broken build

**Review by orenbm — APPROVED** [review]

**Review by sigalsax — APPROVED** [review]
LGTM! great work :)
